src/Mod/Fem/App/FemVTKTools.cpp

```diff
@@ -1063,6 +1064,77 @@ void FemVTKTools::exportFreeCADResult(const App::DocumentObject* result, vtkSmar
     Base::Console().log("End: Create VTK result data from FreeCAD result data.\n");
 }
 
+void FemVTKTools::addArrayFromFunction(
+    vtkSmartPointer<vtkDataObject>& data,
+    const std::map<std::string, std::string>& functions
+)
+{
+    if (!data) {
+        return;
+    }
+
+    auto calculator = vtkSmartPointer<vtkArrayCalculator>::New();
+    std::vector<vtkSmartPointer<vtkDataSet>> fields;
+
+    if (auto dataSet = vtkDataSet::SafeDownCast(data)) {
+        fields.emplace_back(dataSet);
+    }
+    else if (auto blocks = vtkMultiBlockDataSet::SafeDownCast(data)) {
+        for (unsigned int i = 0; i < blocks->GetNumberOfBlocks(); ++i) {
```

Comment:
I don't know if that can be the case here, but `vtkMultiBlock`s can contain other multiblocks. You can use `vtkDataObjectTreeIterator` for iteration instead.